packages/tx/src/legacyTransaction.ts

```diff
@@ -76,6 +76,14 @@ export default class Transaction extends BaseTransaction<Transaction> {
 
     const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
 
+    const integerProps = [nonce, gasPrice, gasLimit, value, v, r, s]
+    for (const x of integerProps) {
+      if (toBuffer(x).length > 0 && x[0] === 0x00) {
```

Comment:
these are already Buffers so they don't need to be routed through toBuffer again

Reply:
Hmm, in some of the API tests, this was failing with a type error related to x.length being undefined but this resolved it since it accounts for blank values.  Is this rather a matter for fixing another test you think?

Reply:
hm, i don't think we should expect a non-Buffer value to be passed to fromValuesArray, it should be formatted beforehand to respect the input param type (`Buffer[]`). if `length` is undefined then a Buffer is probably not being passed in?

Reply:
Okay, let me change it back and look at the test that was failing.

Reply:
It's failing in the [base.spec.ts](https://github.com/ethereumjs/ethereumjs-monorepo/blob/eb5240bec432b5c37f38fd5483483f805f24f234/packages/tx/test/base.spec.ts#L128) because an unsigned transaction was being passed in.  I added a second check for v, r, and s only if they exist in the `txValues` passed in.